paper-api/src/main/java/io/papermc/paper/event/packet/UncheckedSignChangeEvent.java

```diff
@@ -0,0 +1,88 @@
+package io.papermc.paper.event.packet;
+
+import io.papermc.paper.math.BlockPosition;
+import io.papermc.paper.math.Position;
+import net.kyori.adventure.text.Component;
+import org.bukkit.block.sign.Side;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Unmodifiable;
+import org.jspecify.annotations.NullMarked;
+import java.util.List;
+
+/**
+ * Called when a client attempts to modify a sign, but the location at which the sign should be edited
+ * has not yet been checked for the existence of a real sign.
+ * <p>
+ * Cancelling this event will prevent further processing of the sign change, but needs further handling
```

Comment:
In regards to this, is there a point in us just resending the block?
Usage of this event would always be with "ephemeral" signs.

Leaving this up to plugin developers feels pretty meh.

Reply:
I also pondered about doing it that way, but at the end I thought the event should just report back that the player tried to edit a sign and minimize other side effects.

Reply:
I mean, it's unchecked, and so I generally expect it to be a lot more 'dumb' in terms of how it interacts. I feel that such low level events are always going to be kinda meh